float64 linear
float64 angular
bool rect_detected